### PR TITLE
Introduce FormScreenCaptureMode property on Form

### LINKED ISSUE #13258: [API Suggestion][Security] Add ExcludeFromScreenCapture API to Forms/Controls
## TL;DR

With the rise of continuous screen capture tools for AI analysis, sensitive LOB applications are at increased risk of data leakage. This proposal adds a simple API to WinForms that allows developers to prevent their forms and controls from being captured in screenshots, addressing a critical security concern for applications that handle sensitive data.

## Rationale
Modern productivity tools increasingly use continuous screen capturing to feed content to LLMs for summarization, analysis, and automation. While this offers productivity benefits, it creates significant security risks for certain Line of Business (LOB) applications that handle sensitive data, or at least certain areas of those apps.

Here are a few thoughts, which types of applications might require protection against unauthorized screen captures:

1. **Healthcare Applications**: Applications displaying patient records, medical histories, test results, or other protected health information (PHI) covered under HIPAA regulations
   
2. **Financial Applications**: Banking software, investment platforms, or accounting systems showing account balances, transaction histories, credit card numbers, or tax information
   
3. **Government Systems**: Applications handling citizen data, social security numbers, tax records, or classified information
   
4. **Legal Software**: Case management systems containing privileged attorney-client communications, court filings, or sensitive legal strategies
   
5. **Insurance Platforms**: Systems displaying policy details, claims information, or personal policyholder data
   
6. **HR Software**: Applications showing employee records, salary information, performance reviews, or personal details
   
7. **Educational Systems**: Applications containing student records, grades, or other information protected under FERPA

## Proposed API

Windows 10/11 provides the [`SetWindowDisplayAffinity` API](https://learn.microsoft.com/windows/win32/api/winuser/nf-winuser-setwindowdisplayaffinity) with the options `WDA_NONE`, `WDA_Monitor` and `WDA_EXCLUDEFROMCAPTURE` that can block most software-based screen capture methods. This proposal wraps this functionality in a simple, easy-to-use WinForms API that raises awareness about this security concern and gives developers a straightforward way to protect sensitive information.

### New Property for `Form` Class

```csharp
namespace System.Windows.Forms;

public class Form
{
   /// <summary>
   /// Gets or sets a value indicating whether this form should be excluded from screen captures.
   /// When set to true, the form will appear blacked out in screenshots and screen recordings.
   /// This only works on Windows 10 version 1803 and higher.
   /// </summary>
   public ScreenCaptureMode ScreenCaptureMode { get; set; }
}
   ```

### Alternatively: New Property for `Control` Class
[update: we're not doing this. Look in the edits-history for the original idea.]

### Additional Supporting APIs
```csharp
namespace System.Windows.Forms;

/// <summary>
/// Enumeration defining the behavior when a form or control is captured in a screenshot.
/// </summary>
public enum ScreenCaptureMode
{
    /// <summary>
    ///  The form or control can be captured normally in screenshots. Default.
    /// </summary>
    Allow = 0,
    
    /// <summary>
    ///  The form or control appears blacked out in screenshots.
    /// </summary>
    HideContent = 1,
    
    /// <summary>
    ///  The form or control appears blurred in screenshots.
    /// Requires Windows 10 version 2004 or higher.
    /// </summary>
    HideWindow = 2
}
```

### New Form Configuration in Visual Studio Designer
Add a "Security" category in the Properties panel of the WinForms designer with the `ExcludeFromScreenCapture` property for easy discovery and configuration.
- Make sure this feature is not clashing with the Designer. We might need a design-time Shadow Property to not invoke this feature at design time.

## Implementation Notes
- Assumingly pretty straight forward, low risk, low hanging fruit overall with substantial benefit/impact for customers. Need to explore the feasibility of a per-control level. [Update: Not feasable, we are dropping that.]

- **IMPORTANT:** There is a feature which not many people know of, where a whole Form can be used as a control, in which case the Form acts as a UserControl. If we decide, control level for that feature is not feasible, I would even suggest to throw an exception, if a Form is added to the parent's `ControlCollection` with its `ScreenCaptureMode` set to anything else than `Allow`.

## Limitations/Discussion points
- This full Win32 API is available on Windows 10 version 2004 and higher.
- It of course doesn't prevent hardware-based screen captures (e.g., taking a photo of the screen), which need to be clearly stated in the API docs, along, that this can help securing data but not totally prevent it.
- The protection applies at the window level and may not be granular enough for all scenarios. There are also mechanisms to circumvent this, although apps need to do considerably more effort to do so. We will be adding a few respective warnings in the docs.

## Benefits
- Simple API that's easy for developers to adopt
- Raises awareness about an important security consideration
- Provides immediate protection against many common screen capture methods
- Aligns with the "secure by default" principle by making security features easily accessible

### LINKED ISSUE #13450: Let's consider to rename `ScreenCaptureMode` into `FormScreenCaptureMode`
This is related to #13258.

While implementing this property, which by the way works amazingly well for Form, see GIF below, I was wondering:

Image

Since, as I predicted, this is working only on Top-Level Windows, which equals "Form" in WinForms, and we also already have the properties `FormCornerPreference`, `FormBorderColor`, `FormCaptionBackColor`, `FormCaptionTextColor` if we should rename:

`ScreenCaptionMode` -> `FormScreenCaptureMode`

But I _would_ actually rather leave the enum like this:

```
namespace System.Windows.Forms;

/// <summary>
///  Enumeration defining the behavior when a form or control is captured in a screenshot.
/// </summary>
public enum ScreenCaptureMode
{
    /// <summary>
    ///  The form or control can be captured normally in screenshots. Default.
    /// </summary>
    Allow = 0,

    /// <summary>
    ///  The form or control appears blacked out in screenshots.
    /// </summary>
    HideContent = 1,

    /// <summary>
    ///  The form or control appears blurred in screenshots.
    /// Requires Windows 10 version 2004 or higher.
    /// </summary>
    HideWindow = 2
}
```

### PR DESCRIPTION
**Pull request by KlausLoeffelmann**
Fixes #13258.
Fixes #13450.

Looks like this, when _recorded_:

ScreenCapture

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/13561)

### CODE REVIEW

File: src/System.Windows.Forms/System/Windows/Forms/Form.cs [hunk 1]
```diff
@@ -1723,6 +1724,88 @@ internal override Control? ParentInternal
...
+    ///  </para>
+    ///  <para>
+    ///   The setting takes effect immediately for visible forms. For forms that aren't yet shown,
+    ///   the setting takes effect when the form becomes visible.
+    ///  </para>
+    ///  <para>
+    ///   Warning! Additional top-level windows, which will be spinning off the form, will not be affected
+    ///   by this setting. Please be aware, that for example sensitive information in a pop-up window
+    ///   like a MessageBox, a context menu, DropDownButton menus or PullDown-menus can not be prevented
+    ///   from being captured!
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms/System/Windows/Forms/Form.cs]: Don't use `!`.

File: src/System.Windows.Forms/System/Windows/Forms/Form.cs [hunk 2]
```diff
@@ -1723,6 +1724,88 @@ internal override Control? ParentInternal
...
+    ///  </para>
+    ///  <para>
+    ///   Note that this setting only affects the current form and not any other windows in the application.
+    ///   Also note that this will also affect video streaming applications for video conferencing, etc.
+    ///  </para>
+    ///  <para>
+    ///   Disclaimer: Please keep in mind, that this setting can of course not prevent screen recordings by tools,
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms/System/Windows/Forms/Form.cs]: I'd avoid going into too much detail and just call out https://learn.microsoft.com/windows/win32/api/winuser/nf-winuser-setwindowdisplayaffinity#remarks. "This cannot prevent all capturing, see the Windows documentation {} for more details."

File: src/System.Windows.Forms/System/Windows/Forms/ScreenCaptureMode.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    Allow = 0,
+
+    /// <summary>
+    ///  The form or control appears blacked out in screenshots.
+    /// </summary>
+    HideContent = 1,
+
+    /// <summary>
+    ///  The form or control appears blurred in screenshots.
+    /// Requires Windows 10 version 2004 or higher.
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms/System/Windows/Forms/ScreenCaptureMode.cs]: This line is a remark.

File: src/System.Windows.Forms/System/Windows/Forms/Form.cs [hunk 4]
```diff
@@ -1723,6 +1724,88 @@ internal override Control? ParentInternal
...
+        }
+    }
+
+    private bool ShouldSerializeFormScreenCaptureMode() =>
+        FormScreenCaptureMode != ScreenCaptureMode.Allow;
+
+    private void ResetFormScreenCaptureMode() =>
+        FormScreenCaptureMode = ScreenCaptureMode.Allow;
+
+    private void SetScreenCaptureModeInternal(ScreenCaptureMode value)
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms/System/Windows/Forms/Form.cs:1795]: Just do this inline as there is only one caller.

File: src/System.Windows.Forms/System/Windows/Forms/Form.cs [hunk 5]
```diff
@@ -1723,6 +1724,88 @@ internal override Control? ParentInternal
...
+            Properties.AddOrRemoveValue(s_propFormScreenCaptureMode, value);
+
+            if (IsHandleCreated)
+            {
+                SetScreenCaptureModeInternal(value);
+            }
+        }
+    }
+
+    private bool ShouldSerializeFormScreenCaptureMode() =>
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms/System/Windows/Forms/Form.cs:1789]: Ugh, we need to turn on IDE0051 so we don't accidentally lose these designer specific methods. Don't worry about it here, just noting that I thought I had turned it to warning.

**Comment by KlausLoeffelmann** [comment]
> I encourage you to try to let CoPilot to "Rewrite the XML remarks for FormScreenCaptureMode for proper grammar, clarity, reduced duplication of information, and professional tone.". Would be good to come up with a prompt that we can use to make our new docs more polished.
> 
> This also needs some unit tests, in particular negative cases.

I can add that to the Copilot-instructions later I have been working with so far and do a separate PR with it at a later point.

File: src/System.Windows.Forms/System/Windows/Forms/Form.cs (hunk 4 above)
Comment (KlausLoeffelmann) [reply src/System.Windows.Forms/System/Windows/Forms/Form.cs:1795]: Oh, and that will be actually a problem!!
Fixed! (But differently.)

Comment (KlausLoeffelmann) [reply src/System.Windows.Forms/System/Windows/Forms/Form.cs:1795]: This is also called now from `ReCreateHandle`.

File: src/System.Windows.Forms/System/Windows/Forms/Form.cs (hunk 2 above)
Comment (KlausLoeffelmann) [reply src/System.Windows.Forms/System/Windows/Forms/Form.cs]: Yep, rewrote per your prompt request.

File: src/System.Windows.Forms/System/Windows/Forms/Form.cs (hunk 5 above)
Comment (KlausLoeffelmann) [reply src/System.Windows.Forms/System/Windows/Forms/Form.cs:1789]: Because of no obvious call reference you mean?

Comment (JeremyKuhne) [reply src/System.Windows.Forms/System/Windows/Forms/Form.cs:1789]: @KlausLoeffelmann yeah, because of that. Enforcing this is super useful for finding actual dead code.

**Review by JeremyKuhne — APPROVED** [review]
Looks good.

**Review by Shyam-Gupta — DISMISSED** [review]

File: src/System.Windows.Forms/Resources/SR.resx [hunk 6]
```diff
@@ -7033,10 +7033,16 @@ Stack trace where the illegal operation occurred was:
...
+  <data name="ClipboardOrDragDrop_CannotJsonSerializeDataObject" xml:space="preserve">
     <value>An object of type 'DataObject' will serialize as empty. Use 'DataObject.SetDataAsJson' APIs on your object to JSON-serialize the data within your object, then use the '{0}' API with your object.</value>
   </data>
-  <data name="NotSupportedExceptionOnClipboard">
+  <data name="NotSupportedExceptionOnClipboard" xml:space="preserve">
     <value>Clipboard contains a 'NotSupportedException' that indicates that write operation failed. '{0}'.</value>
   </data>
-</root>
+  <data name="FormScreenCaptureModeDescr" xml:space="preserve">
+    <value>Gets or sets the behavior for preventing screen capture/video recording of the form's content via screen capture applications based on the Windows APIs.</value>
```
Comment (Shyam-Gupta) [inline src/System.Windows.Forms/Resources/SR.resx:7043]: Should 'F' be in uppercase in form's ?
